src/server/req_manager.c

```diff
@@ -2521,49 +2783,58 @@ create_pbs_node2(char *objname, svrattrl *plist, int perms, int *bad, struct pbs
 
 		/* need to create the pbs_node entry */
 
-		/* find an empty slot or make a new one */
-		for (iht=0; iht<svr_totnodes; iht++) {
-			if (pbsndlist[iht]->nd_state & INUSE_DELETED) {
-				/*available, use*/
-				pnode = pbsndlist[iht];
-				break;
-			}
+		pnode =(struct pbsnode *)malloc(sizeof(struct pbsnode));
+		if (pnode == (struct pbsnode *)0) {
+			free(pname);
+			return (PBSE_SYSTEM);
 		}
 
-		if (iht == svr_totnodes) {    /*no unused entry, make an entry*/
-
-			pnode =(struct pbsnode *)malloc(sizeof(struct pbsnode));
-			if (pnode == (struct pbsnode *)0) {
-				free(pname);
-				return (PBSE_SYSTEM);
-			}
-
-			/* expand pbsndlist array exactly svr_totnodes long*/
-			tmpndlist = (struct pbsnode **)realloc(pbsndlist,
-				sizeof(struct pbsnode*) * (svr_totnodes + 1));
+		/* expand pbsndlist array exactly svr_totnodes long*/
+		tmpndlist = (struct pbsnode **)realloc(pbsndlist,
+			sizeof(struct pbsnode*) * (svr_totnodes + 1));
 
-			if (tmpndlist != (struct pbsnode **)0) {
-				/*add in the new entry etc*/
-				pbsndlist = tmpndlist;
-				pnode->nd_index = svr_totnodes;
-				pbsndlist[svr_totnodes++] = pnode;
-			} else {
-				free(pnode);
-				free(pname);
-				return (PBSE_SYSTEM);
-			}
+		if (tmpndlist != (struct pbsnode **)0) {
+			/*add in the new entry etc*/
+			pbsndlist = tmpndlist;
+			pnode->nd_index = svr_totnodes;
+			pnode->nd_arr_index = svr_totnodes; /* this is only in mem, not from db */
+			pbsndlist[svr_totnodes++] = pnode;
+		} else {
+			free_pnode(pnode);
+			free(pname);
+			return (PBSE_SYSTEM);
 		}
 		if (initialize_pbsnode(pnode, pname, ntype) != PBSE_NONE) {
-			set_vnode_state(pnode, INUSE_DELETED, Nd_State_Set);
+			svr_totnodes--;
+			free_pnode(pnode);
 			free(pname);
 			return (PBSE_SYSTEM);
 		}
 
 		/* create and initialize the first subnode to go with */
 		/* the parent node */
-
 		if (create_subnode(pnode, NULL) == NULL) {
-			set_vnode_state(pnode, INUSE_DELETED, Nd_State_Set);
+			svr_totnodes--;
+			free_pnode(pnode);
+			free(pname);
+			return (PBSE_SYSTEM);
+		}
+
+		/* create node tree if not already done */
+		if (node_tree == NULL ) {
+			node_tree = create_tree(0, 0);
```

Comment:
done.  I checked for other instances, and I think we've found them all now.  Thanks.